File: Data Structures & Algorithms/two-integer-sum/submission-3.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> hashMap;

        for (int i = 0; i < nums.size(); i++) {
            int num = nums[i];
            int complement = target - num;
            if (hashMap.find(complement) != hashMap.end()) {
                return { hashMap[complement], i };
            }
            hashMap.insert({ num, i });
        }

        return {};
    }
};

File: Data Structures & Algorithms/binary-search/submission-3.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        st, end = 0, len(nums) - 1

        while st <= end:
            mid = st + (end - st)//2
            if target < nums[mid]:
                end = mid - 1
            elif target > nums[mid]:
                st = mid + 1
            else:
                return mid
        
        return -1